Capstone/Assets/Game/Scripts/Player.cs

```diff
@@ -687,5 +714,64 @@ public override bool TakeDamage(int damagingID, float damage)
             // Return false if the player cannot currently be damaged
             return false; 
         }
+
+        public override void Respawn()
+        {
+            // If the player's health hasn't been reset yet
+            if (Health != MaxHealth)
+            {
+                // Do one time only resets
+                Health = MaxHealth;
+                transform.position = respawnPoint;
+                transform.rotation = Quaternion.Euler(respawnRotation);
+                cinemachineTargetYaw = respawnRotation.y;
+                cinemachineTargetPitch = respawnRotation.z;
+            }
+
+            // Call the fade in method multiple times so it can fade
+            FadeIn();
+
+            // When the fade in is done, change current state
+            if(fade.GetComponent<Image>().color.a <= 0)
+            {
+                FSM.SetCurrentState(idleCombat);
+                FSM.SetCurrentState(idleMovement);
+            }
+        }
+
+        private void FadeToBlack()
+        {
+            // Unhide the fade out image
+            if (fade.activeSelf == false)
```

Comment:
It's a small thing but I think we mentioned doing brackets around one line if statements.